packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -202,6 +203,42 @@ export class BlobManager {
         });
     }
 
+    /**
+     * Generates data used for garbage collection. Each blob uploaded represents a node in the GC graph as it can be
+     * individually referenced by storing its handle in a referenced DDS. Returns the list of blob ids as GC nodes.
+     * @param fullGC - true to bypass optimizations and force full generation of GC data. BlobManager doesn't care
+     * about this for now because the data is a simple list of blob ids.
+     */
+    public getGCData(fullGC: boolean = false): IGarbageCollectionData {
+        const gcData: IGarbageCollectionData = { gcNodes: {} };
+        /**
+          * The GC node id is of the format `/_blobs/blobId`. This node is must match the path of the blob handle
+          * returned by the createBlob API because blobs are marked referenced by storing these handles in a referenced
+          * DDS. There are no outbound routes from these nodes as it  can only be referenced by other nodes in the
+          * container but cannot referenced other nodes.
+          */
+        this.blobIds.forEach((blobId: string) => {
+            gcData.gcNodes[`/${BlobManager.basePath}/${blobId}`] = [];
```

Comment:
@wes-carlson , can you please check if this looks correct?
Basically the goal of this function is to enumerate all possible handles that blob manager ever created, and paths of these handles. We compare that list against actual handles stored in DDSs to find any paths that are not referenced in a file, and thus blobs can be deleted.

We need to ensure this list is correct and takes into account any remappings that happen in blob manager.

Reply:
Ideally, this only runs in the summarizer so the container would not be detached. But maybe we should make this flexible so it works irrespective of that? So, maybe make this similar to snapshot which looks at pending blobs?

Reply:
Looking at getBlob() implementation, I think this code (and code below) should consult this.redirectTable. All blob IDs (keys) are actually used by handles. And any values that it points to (values) should be not considered (removed) when consulting this.blobIds.

Reply:
Yeah, looks like it.

Reply:
I believe its a little more complex than simply using the redirect table. If a blob is in the redirect table, it has a local id and a storage id. The blob could be referenced by either the local id or storage id or both - In detached container, you get the local id and in attached, you get the storage id.
So, these blobs have 2 GC nodes (one for local id and one for storage id) and if either of them is referenced, the blob should not be marked unreferenced.

Reply:
@wes-carlson to confirm this.

Reply:
I'm not sure about how the GC works, but yes a blob might be referenced only by handles with local IDs (in the future there may only be local IDs in handles). In this case it should not be considered unreferenced.

There are some tests here if you want to add some similar GC tests for local ID blob handles https://github.com/microsoft/FluidFramework/blob/0cfeab59012500baf0c56369795d42378fc46845/packages/test/test-end-to-end-tests/src/test/blobs.spec.ts#L237

Reply:
Okay. So, it seems that blobs created in detached mode are only represented by localId outside of the blob manager. Basically, the handle returned has localId and there is no way to get a handle with the storageId.
I added code to handle scenario where the localId handle is added to a DDS and it should mark the storageId blob as referenced. Also added tests to blobs.spec.ts to validates these scenarios.

Reply:
> Basically, the handle returned has localId and there is no way to get a handle with the storageId.

I think if the same blob were uploaded, it could be deduped and we would get a handle with the storage ID. I believe all services including local server do deduping, so it should be possible to test a scenario like this.

Reply:
Good catch. I handled that scenario and added tests for it.
Another scenario that is possible is that multiple localId redirects to the same storageId. This can happen when uploading the same blob content in detached mode and the detached storage manager doesn't de-dup blobs. Later when these blobs are uploaded, they will be de-duped by the server. Added tests for this scenario as well.

The logic for GC blob manager with localIds when sweep will run (currently implemented by `detelteUnusedRoutes` function which runs in GC test mode:
- Any used localId will mark the corresponding storageId blob as referenced.
- Any unused localId will be removed from the redirect table. The corresponding storageId may still be used.
- Any unused blobId (including storageId) will be deleted and will be gone from summary.
- When multiple localIds redirect to the same storageId, if one of the localIds is used, the storageId is referenced and unused localIds are removed from the redirect table.